### PR TITLE
Add options to run unit tests to TensorFlow EasyBlock

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyblocks/t/tensorflow.py [hunk 1]
```diff
@@ -636,9 +640,6 @@ def configure_step(self):
                                                                              self.install_base_dir))]
             apply_regex_substitutions('configure.py', regex_subs)
 
-        # Tell Bazel to not use $HOME/.cache/bazel at all
```
Comment (bartoldeman) [inline easybuild/easyblocks/t/tensorflow.py:639]: This is removed without justification and looks unrelated.

**Review by bartoldeman — DISMISSED** [review]
Apart from the mystery removal this lgtm

File: easybuild/easyblocks/t/tensorflow.py (hunk 1 above)
Comment (Flamefire) [reply easybuild/easyblocks/t/tensorflow.py:639]: Yes, I'm currently testing why the tests still fail (they are run at least) and found that we add to many options. Only thing needed seems to be `--output_user_root`, TEST_TMPDIR is set internally by Bazel to tell Bazel that a test is being run. Setting that for compilation is hence wrong. I'll double check if output_base and install_base can also be removed and would include that here, or shall I make a second PR where those options are removed? Problem then would be merge conflicts which I'd want to avoid.

Comment (bartoldeman) [reply easybuild/easyblocks/t/tensorflow.py:639]: A second PR would be best indeed.
Perhaps a rebase can avoid merge conflicts?

Comment (Flamefire) [reply easybuild/easyblocks/t/tensorflow.py]: Let's see. I undid this and if this is merged first I can keep the other changes and likely rebase them

**Comment by Flamefire** [comment]
@bartoldeman *ping*

File: easybuild/easyblocks/t/tensorflow.py [hunk 2]
```diff
@@ -222,6 +222,12 @@ def extra_options():
             'with_mkl_dnn': [with_mkl_dnn_default, "Make TensorFlow use Intel MKL-DNN", CUSTOM],
             'with_xla': [None, "Enable XLA JIT compiler for possible runtime optimization of models", CUSTOM],
             'test_script': [None, "Script to test TensorFlow installation with", CUSTOM],
+            'test_targets': [[], "List of Bazel targets which should be run during the test step", CUSTOM],
+            'test_tag_filters': ['', "Comma-separated list of tags to filter for during the test step", CUSTOM],
+            'test_jobs': [None, "Number of test jobs to run in parallel. Use None to automatically determine a value",
```
Comment (akesandgren) [inline easybuild/easyblocks/t/tensorflow.py:227]: Should perhaps add that None is the default in the help text here.

File: easybuild/easyblocks/t/tensorflow.py [hunk 3]
```diff
@@ -719,43 +725,50 @@ def build_step(self):
...
-            'bazel',
+        # Options passed to the bazel command
+        self.bazel_opts = [
             '--output_base=%s' % self.output_base_dir,
             '--install_base=%s' % self.install_base_dir,
             '--output_user_root=%s' % self.output_user_root_dir,
-            'build',
         ]
+        jvm_max_memory = self.cfg['jvm_max_memory']
+        if self.cfg['jvm_max_memory']:
```
Comment (akesandgren) [inline easybuild/easyblocks/t/tensorflow.py:735]: Use the variable you just assigned instead of repeating the self.cfg ?

File: easybuild/easyblocks/t/tensorflow.py (hunk 2 above)
Comment (Flamefire) [reply easybuild/easyblocks/t/tensorflow.py]: Done

File: easybuild/easyblocks/t/tensorflow.py (hunk 3 above)
Comment (Flamefire) [reply easybuild/easyblocks/t/tensorflow.py]: Good catch, thanks!

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @Flamefire!
